src/commands/upload_file.ts

```diff
@@ -121,6 +124,42 @@ function getSingleFile(parameters: ParametersHelper, parentFolderId: FolderID):
 	return [singleParameter];
 }
 
+async function getRemoteFile(
+	parameters: ParametersHelper,
+	parentFolderId: FolderID
+): Promise<UploadPathParameter[] | undefined> {
+	const remoteFilePath = parameters.getParameterValue(RemotePathParameter);
+
+	if (!remoteFilePath) {
+		return undefined;
+	}
+
+	const tempFolder = getTempFolder();
+	const destinationFileName = parameters.getRequiredParameterValue(DestinationFileNameParameter);
+
+	const { pathToFile, contentType } = await download(
+		remoteFilePath,
+		tempFolder,
+		destinationFileName,
+		(downloadProgress: number) => {
+			if (showProgressLog) {
+				process.stderr.write(`Downloading file... ${downloadProgress}\r`);
+			}
+		}
+	);
```

Comment:
How often does this trigger, do we need a debounce involved here?

Reply:
From the looks of your demo video, very frequently. Could be worth debouncing it to achieve something like 3fps max. Non-blocking.

Reply:
It is trigger for each downloaded chunk